src/api/types.ts

```diff
@@ -81,44 +81,77 @@ export interface Receipt {
   OWL(): Promise<Account>,
   approvedTokens(address: Account): Promise<boolean>,
   latestAuctionIndices(token1: Account, token2: Account): Promise<BigNumber>,
   auctionStarts(token1: Account, token2: Account): Promise<BigNumber>,
   closingPrices(token1: Account, token2: Account, index: Index): Promise<[BigNumber, BigNumber]>,
-  balances(token: Account, account: Account): Promise<BigNumber>,
-  sellVolumes(token1: Account, token2: Account, index: Index): Promise<BigNumber>,
   sellVolumesCurrent(token1: Account, token2: Account): Promise<BigNumber>,
   sellVolumesNext(token1: Account, token2: Account): Promise<BigNumber>,
-  buyVolumes(token1: Account, token2: Account, index: Index): Promise<BigNumber>,
-  extraSellTokens(token1: Account, token2: Account, index: Index): Promise<BigNumber>,
-  extraBuyTokens(token1: Account, token2: Account, index: Index): Promise<BigNumber>,
+  buyVolumes(token1: Account, token2: Account): Promise<BigNumber>,
+  balances(token: Account, account: Account): Promise<BigNumber>,
+  extraTokens(token1: Account, token2: Account, index: Index): Promise<BigNumber>,
   sellerBalances(token1: Account, token2: Account, index: Index, account: Account): Promise<BigNumber>,
   buyerBalances(token1: Account, token2: Account, index: Index, account: Account): Promise<BigNumber>,
-  claimedBalances(token1: Account, token2: Account, index: Index, account: Account): Promise<BigNumber>,
+  claimedAmounts(token1: Account, token2: Account, index: Index, account: Account): Promise<BigNumber>,
+  
+  isInitialised(): Promise<boolean>,
 
   NewDeposit: ContractEvent,
   NewWithdrawal: ContractEvent,
   NewSellOrder: ContractEvent,
   NewBuyOrder: ContractEvent,
   NewSellerFundsClaim: ContractEvent,
   NewBuyerFundsClaim: ContractEvent,
+  NewTokenPair: ContractEvent,
   AuctionCleared: ContractEvent,
+  Log: ContractEvent,
+  LogOutstandingVolume: ContractEvent,
+  LogNumber: ContractEvent,
+  ClaimBuyerFunds: ContractEvent,
   allEvents(filter: Filter, cb: ErrorFirstCallback): void,
   allEvents(filter: Filter): EventInstance,
 
-  updateOwner(account: Account, tx?: TransactionObject): Promise<Receipt>,
+  setupDutchExchange(
+    tokenTUL: Account,
+    tokenOWL: Account,
+    auctioneer: Account,
+    tokenETH: Account,
+    ethUSDOracle: Account,
+    thresholdNewTokenPair: Balance,
+    thresholdNewAuction: Balance,
+    tx?: TransactionObject,
+  ): Promise<Receipt>,
+  updateExchangeParams(
+    auctioneer: Account,
+    ethUSDOracle: Account,
+    thresholdNewTokenPair: Balance,
+    thresholdNewAuction: Balance,
+    tx?: TransactionObject,
+  ): Promise<Receipt>,
   updateApprovalOfToken(token: Account, approved: boolean, tx?: TransactionObject): Promise<Receipt>,
-  updateETHUSDPriceOracle(_ETHUSDOracletoken: Account, tx?: TransactionObject): Promise<Receipt>,
+  startMasterCopyCountdown(masterCopy: Account, tx?: TransactionObject): Promise<Receipt>,
+  updateMasterCopy(tx?: TransactionObject): Promise<Receipt>,
```

Comment:
For some of these `tx` parameter isn't optional as `msg.sender` is checked inside the function.

Reply:
hmm thought I had changed that. Thanks